bazaarvoice/bv-ui-core
# JST-7: Add region module

## What's in this PR ?

New laws in the EU require that identifying information stay physically located in the EU. In order to comply with this, we'll need to be able to tell when we're running in the EU. It has been determined that the onus of providing the correct locale is on the client, so we can safely rely on the base locale of an implementation.

This PR adds a module that UI projects can use to determine if it's in EU or not. It has been designed with extensibility in mind, in case we want to add support for detecting other regions in the future.

## How to verify

1. Check out this PR
1. In the `bv-ui-core` repo, run `node`.
1. Run `var region = require('./lib/region');` to get a reference to the root module.
1. Run `region.hasLocale('europe', 'en_US');`. Should return `false`.
1. Run `region.hasTerritory('europe', 'US');`. Should return `false`.
1. Run `region.hasLocale('europe', 'en_GB');`. Should return `true`.
1. Run `region.hasTerritory('europe', 'GB');`. Should return `true`.
1. Run `region.lookupLocale('en_GB'); Should return `'europe'`.
1. Run `region.lookupLocale('en_US'); Should return `'unsupported'`.
1. If you prefer, instead of the "generic" module, you can run `var europe = require('./lib/region/europe');` to load the European specific module. Useful if you're trying to keep file size down and aren't worried about detecting other regions in the future.
1. Feel free to try it out with a bunch of different locales. The list of country codes provided in this PR is intended to be the official EU list. If you want more things to try, look at the unit tests for examples.

## Unit tests

So long as you have the npm modules installed, you can run the unit tests with a simple `npm test`.
